### PR TITLE
switch to new csproj

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: exercises/01-composite-ui/after/Divergent.Frontend/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  },
+  "profiles": {
+    "IIS Express": {
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
```
Comment (mauroservienti) [inline exercises/01-composite-ui/after/Divergent.Frontend/Properties/launchSettings.json]: this could be simplified by removing all what is related to IIS Express, we just run console apps to host Kestrel.

File: exercises/01-composite-ui/before/Divergent.Frontend/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  },
+  "profiles": {
+    "IIS Express": {
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
```
Comment (mauroservienti) [inline exercises/01-composite-ui/before/Divergent.Frontend/Properties/launchSettings.json]: this could be simplified by removing all what is related to IIS Express, we just run console apps hosting Kestrel

File: exercises/02-publish-subscribe/after/Divergent.Frontend/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  },
+  "profiles": {
+    "IIS Express": {
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
```
Comment (mauroservienti) [inline exercises/02-publish-subscribe/after/Divergent.Frontend/Properties/launchSettings.json]: this could be simplified by removing all what is related to IIS Express, we just run console apps hosting Kestrel

File: exercises/02-publish-subscribe/before/Divergent.Frontend/Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    }
+  },
+  "profiles": {
+    "IIS Express": {
+      "commandName": "IISExpress",
+      "launchBrowser": true,
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
```
Comment (mauroservienti) [inline exercises/02-publish-subscribe/before/Divergent.Frontend/Properties/launchSettings.json]: this could be simplified by removing all what is related to IIS Express, we just run console apps hosting Kestrel

**Comment by mauroservienti** [comment]
In theory we could also get rid of all the `<LangVersion>latest</LangVersion>` in all projects by adding a `Directory.Build.props` file with that configuration.

**Comment by adamralph** [comment]
@mauroservienti good call on the IIS stuff. I'll kill it.

> In theory we could also get rid of all the `<LangVersion>latest</LangVersion>` in all projects by adding a `Directory.Build.props` file with that configuration.

Good idea, although it's orthogonal to this PR (the lines containing `<LangVersion>latest</LangVersion>` are unchanged).

**Comment by adamralph** [comment]
@mauroservienti the PR is now ready for full review.

File: exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj [hunk 5]
```diff
@@ -1,72 +1,19 @@
...
-    <Compile Include="Models\Order.cs" />
-    <None Include="App.config" />
-    <None Include="packages.config" />
+    <ProjectReference Include="..\Divergent.ITOps.Interfaces\Divergent.ITOps.Interfaces.csproj" />
   </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+
   <PropertyGroup>
-    <PostBuildEvent>copy /Y "$(TargetDir)$(ProjectName).dll" "$(SolutionDir)Divergent.ITOps\Providers\$(ProjectName).dll"</PostBuildEvent>
+    <PostBuildEvent>copy /Y "$(TargetDir)Divergent.Customers.Data.dll" "$(SolutionDir)Divergent.ITOps\Providers\Divergent.Customers.Data.dll"</PostBuildEvent>
```
Comment (mauroservienti) [inline exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj]: Was this change required because of the bug in post build events with `$(ProjectName)`?

File: exercises/04-integration/before/Divergent.Customers.Data/Divergent.Customers.Data.csproj [hunk 6]
```diff
@@ -1,72 +1,19 @@
...
-    <None Include="App.config" />
-    <None Include="packages.config" />
-    <Folder Include="ITOps\" />
+    <ProjectReference Include="..\Divergent.ITOps.Interfaces\Divergent.ITOps.Interfaces.csproj" />
   </ItemGroup>
-  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+
   <PropertyGroup>
-    <PostBuildEvent>copy /Y "$(TargetDir)$(ProjectName).dll" "$(SolutionDir)Divergent.ITOps\Providers\$(ProjectName).dll"</PostBuildEvent>
+    <PostBuildEvent>copy /Y "$(TargetDir)Divergent.Customers.Data.dll" "$(SolutionDir)Divergent.ITOps\Providers\Divergent.Customers.Data.dll"</PostBuildEvent>
```
Comment (mauroservienti) [inline exercises/04-integration/before/Divergent.Customers.Data/Divergent.Customers.Data.csproj]: see my previous comment

File: exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj (hunk 5 above)
Comment (adamralph) [reply exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj]: @mauroservienti for one reason or another, `$(ProjectName)` is empty when using SDK csproj. I don't know if this is a bug or by design, but I thought the easiest way to work around it is to hardcode the project name. It's unlikely to change and if it ever does, we'll see a problem straight away when smoke testing the exercise.

BTW - I had to do the same thing in `Divergent.Shipping.Data.csproj`.

**Comment by adamralph** [comment]
@danielmarbach @dvdstelt it's up to you whether we merge this before or after SELA.

File: exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj (hunk 5 above)
Comment (mauroservienti) [reply exercises/04-integration/after/Divergent.Customers.Data/Divergent.Customers.Data.csproj]: It’s a bug, I faced it a few months ago, basically with the new project style not all variables are filled correctly before the execution of pre/post build steps.

**Comment by mauroservienti** [comment]
Removing myself from the assignees as this is done, and it’s only waiting to be merged.
